src/auth/src/token_cache.rs

```diff
@@ -26,23 +27,43 @@ use tokio::time::{Duration, Instant, sleep};
 const NORMAL_REFRESH_SLACK: Duration = Duration::from_secs(240);
 const SHORT_REFRESH_SLACK: Duration = Duration::from_secs(10);
 
-#[derive(Debug, Clone)]
-pub(crate) struct TokenCache {
+pub(crate) struct TokenCache<T: TokenProvider> {
     rx_token: watch::Receiver<Option<Result<(Token, EntityTag)>>>,
+    token_provider: Arc<T>,
 }
 
-impl TokenCache {
-    pub(crate) fn new<T>(inner: T) -> Self
-    where
-        T: TokenProvider + Send + Sync + 'static,
-    {
+impl<T: TokenProvider> Clone for TokenCache<T> {
+    fn clone(&self) -> Self {
+        Self {
+            rx_token: self.rx_token.clone(),
+            token_provider: self.token_provider.clone(),
+        }
+    }
+}
```

Comment:
Can you use `#[derive(Clone)]` for this? If not, do you mind adding a comment explaining why?

Reply:
Added explanation.